### PR TITLE
Allow to undo muting the chat/channel from toast

### LINKED ISSUE #19058: Fix chat's mute sheet and toast
# Bug Report

## Problem 
Fix according to the [design review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=8065:58901&mode=design#645047172)

- fix sheet header
- fix mute toast styles
- add undo timer to mute toast

### PR DESCRIPTION
**Pull request by clauxx**
fixes #19058

### Summary

Added the undo action to the toast shown when muting the chat/channel.

#### Platforms

- Android
- iOS

##### Functional

- 1-1 chats
- public chats
- group chats

### Steps to test

- Open Status
- Open chat
- Press "Mute chat/channel"
- Press "Undo" on the toast

status: ready

### CODE REVIEW

File: translations/en.json [hunk 1]
```diff
@@ -2311,19 +2311,19 @@
...
+    "channel-muted-for-15-minutes": "Channel muted for 15 minutes\n(until {{duration}})",
+    "channel-muted-for-1-hour": "Channel muted for 1 hour\n(until {{duration}})",
+    "channel-muted-for-8-hours": "Channel muted for 8 hours\n(until {{duration}})",
+    "channel-muted-for-1-week": "Channel muted for 1 week \n(until {{duration}})",
+    "channel-muted-till-unmuted": "Channel muted till unmuted\n(until {{duration}})",
+    "chat-muted-for-15-minutes": "Chat muted for 15 minutes\n(until {{duration}})",
+    "chat-muted-for-1-hour": "Chat muted for 1 hour\n(until {{duration}})",
+    "chat-muted-for-8-hours": "Chat muted for 8 hours\n(until {{duration}})",
+    "chat-muted-for-1-week": "Chat muted for 1 week\n(until {{duration}})",
+    "chat-muted-till-unmuted": "Chat muted till unmuted\n(until {{duration}})",
```
Comment (BalogunofAfrica) [inline translations/en.json]:
```suggestion
    "channel-muted-for-15-minutes": "Channel muted for 15 minutes \n(until {{duration}})",
    "channel-muted-for-1-hour": "Channel muted for 1 hour \n(until {{duration}})",
    "channel-muted-for-8-hours": "Channel muted for 8 hours \n(until {{duration}})",
    "channel-muted-for-1-week": "Channel muted for 1 week \n(until {{duration}})",
    "channel-muted-till-unmuted": "Channel muted till unmuted \n(until {{duration}})",
    "chat-muted-for-15-minutes": "Chat muted for 15 minutes \n(until {{duration}})",
    "chat-muted-for-1-hour": "Chat muted for 1 hour \n(until {{duration}})",
    "chat-muted-for-8-hours": "Chat muted for 8 hours \n(until {{duration}})",
    "chat-muted-for-1-week": "Chat muted for 1 week \n(until {{duration}})",
    "chat-muted-till-unmuted": "Chat muted till unmuted \n(until {{duration}})",
```

Having a space before the new line makes it more legible

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/contexts/chat/events.cljs [hunk 2]
```diff
@@ -366,6 +370,27 @@
                       :on-success #(rf/dispatch [:chat/mute-successfully chat-id % mute-type
                                                  muted? chat-type])}]}))
 
+(rf/reg-event-fx
+ :chat.ui/undo-mute
+ (fn [_ [chat-id]]
+   {:fx [[:json-rpc/call
+          [{:method     "wakuext_unmuteChat"
+            :params     [chat-id]
+            :on-error   #(rf/dispatch [:chat/mute-failed chat-id false %])
```
Comment (ibrkhalil) [inline src/status_im/contexts/chat/events.cljs:362]: Should this be: 'Unmute failed'?

**Review by ibrkhalil — APPROVED** [review]
Great work!

File: src/status_im/contexts/chat/events.cljs (hunk 2 above)
Comment (clauxx) [reply src/status_im/contexts/chat/events.cljs:362]: the `:chat/mute`, `:chat/mute-successfully` and `:chat/mute-failed` events handle both muting and unmuting. It should be called toggle really, but idk if it makes sense to be changed in this PR.

**Review by BalogunofAfrica — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
Hi @clauxx thank you for PR. Take a look the found issues:

## ISSUE 1: The "undo" mark is not shown when the community is muted

### Steps:
1. Create community
2. Mute the community

### Actual result:
The "undo" mark is not shown on when the community is muted

https://github.com/status-im/status-mobile/assets/52490791/232b0faf-2d65-4f50-8e6e-cc7795048e0f


### Expected result:
The "undo" mark is shown on the banner when community is muted
https://www.figma.com/file/JlpPhJp0SMnEyBQy4nOZHo/Settings-for-Mobile?type=design&node-id=7099-83572&mode=design&t=hFaG2l7XmpJGSjZ5-0

**Comment by VolodLytvynenko** [comment]
## ISSUE 2: The "undo" mark is not shown when the any chat or community is unmuted

### Steps:
1. Go to any chat
2. Mute the chat
3. Unmute the chat

### Actual result:
The "undo" mark is not shown when the chat is unmuted

https://github.com/status-im/status-mobile/assets/52490791/fa495a7b-fd84-4081-86c1-92ab0fbd664a



### Expected result:
The "undo" mark is shown on the banner when chat is unmuted
https://www.figma.com/file/JlpPhJp0SMnEyBQy4nOZHo/Settings-for-Mobile?type=design&node-id=7099-82451&mode=design&t=hFaG2l7XmpJGSjZ5-0

**Review by jo-mut — APPROVED** [review]

**Comment by clauxx** [comment]
@VolodLytvynenko 
> _ISSUE 1: The "undo" mark is not shown when the community is muted_

Do you know what the behavior would be like here? Say the chat was initially muted 5 days ago for a week. Now they press unmute, the toast appears and they change their mind. They press `undo` and ... is the chat muting resumed (2 more days) or is it muted again for another week? I guess the first variant, but want to make sure as that would take a bit more work.

**Comment by VolodLytvynenko** [comment]
> @VolodLytvynenko
> 
> > _ISSUE 1: The "undo" mark is not shown when the community is muted_
> 
> Do you know what the behavior would be like here? Say the chat was initially muted 5 days ago for a week. Now they press unmute, the toast appears and they change their mind. They press `undo` and ... is the chat muting resumed (2 more days) or is it muted again for another week? I guess the first variant, but want to make sure as that would take a bit more work.

Hey @clauxx, I totally agree with you regarding the first option. The remaining days/hours/minutes should be resumed during which the chat should be muted after 'undo' is tapped. That makes sense

**Comment by clauxx** [comment]
@VolodLytvynenko we had a discussion on discord about this task and it seems like we're adding unnecessary complexity with undo-ing the unmute. I guess we can either close this PR (and leave a comment about it in the issue) or merge the undo for the "mute" action only. WDYT?

**Comment by VolodLytvynenko** [comment]
> @VolodLytvynenko we had a discussion on discord about this task and it seems like we're adding unnecessary complexity with undo-ing the unmute. I guess we can either close this PR (and leave a comment about it in the issue) or merge the undo for the "mute" action only. WDYT?

@clauxx Got it. I agree with you. Lets leave the undo only for 'mute' action.  Could you please only resolve [issue 1](https://github.com/status-im/status-mobile/pull/19074#issuecomment-1988430146)?

**Comment by clauxx** [comment]
@VolodLytvynenko fixed issue 1. Please have a look

**Comment by VolodLytvynenko** [comment]
Hi @clauxx thank you for PR. No issues from my side. PR is ready to be merged

**Pull request merged**